spec/services/course/duplication/course_duplication_service_spec.rb

```diff
@@ -498,6 +498,33 @@
           expect(all_new_comment_emails_disabled).to be(true)
         end
       end
+
+      context 'when sidebar settings have multiple assessment categories' do
+        let!(:categories) { create_list(:course_assessment_category, 2, course: course) }
+
+        before do
+          # Assign sidebar settings for original course by setting the weight to the old
+          # category ID.
+          course.assessment_categories.each do |category|
+            course.settings(:sidebar).settings("assessments_#{category.id}").weight = category.id
+          end
+        end
+
+        it 'updates the sidebar setting keys with the new assessment category IDs' do
+          weights = course.assessment_categories.map(&:id)
+          # Check that sidebar keys with the new assessment category IDs are present and have
+          # values in the original set of weights.
+          new_course.assessment_categories.each do |category|
+            expect(weights).
+              to include(course.settings(:sidebar).settings("assessments_#{category.id}").weight)
+          end
+          # Check that sidebar keys with the old assessment category IDs are not in the new course's
+          # sidebar settings.
+          course.assessment_categories.each do |old_category|
+            expect(course.settings(:sidebar, "assessments_#{old_category.id}").weight).to be_nil
```

Comment:
Need not check the weight, but rather `course.settings(:sidebar, "assessments_#{old_category.id}").to be_nil`

Reply:
Weight is necessary here, or it'll return the `Settings` object.

Reply:
Ok, apologies, verified it on console myself as well 👍